main.go

```diff
@@ -123,6 +125,19 @@ func main() {
 
 	var err error
 
+	//nolint:reassign // We need to reassign ErrorHandlers to register our handler
+	utilruntime.ErrorHandlers = append(utilruntime.ErrorHandlers, func(err error) {
+		var unknownAuthorityError x509.UnknownAuthorityError
+		if errors.As(err, &unknownAuthorityError) {
+			logger.Errorf(err, "Unknown authority error: %v", unknownAuthorityError)
+		}
+
+		var invalidCertificateError x509.CertificateInvalidError
+		if errors.As(err, &invalidCertificateError) {
+			logger.Errorf(err, "Invalid certificate error: %v", invalidCertificateError)
```

Comment:
Using `%v`, we just get the error string printed:

`Invalid certificate error: x509: certificate has expired or is not yet valid: more detail`

which will be printed anyway.

I assume you want the detailed fields printed. With `%#v`, we get:

`Invalid certificate error: x509.CertificateInvalidError{Cert:(*x509.Certificate)(0xc0003f0580), Reason:1, Detail:"more detail"}`

However this does not follow/expand pointers. 

Converting to JSON does (using `resource.ToJSON(err)`):

```
Invalid certificate error: {
  "Cert": {
    "Raw": "AQIDBA==",
    "RawTBSCertificate": null,
    "RawSubjectPublicKeyInfo": null,
    "RawSubject": "BQY=",
    "RawIssuer": null,
    "Signature": null,
    "SignatureAlgorithm": 7,
    "PublicKeyAlgorithm": 2,
    "PublicKey": "PublicKey",
    "Version": 1,
    "SerialNumber": 123,
    "Issuer": {
      "Country": [
        "US"
      ],
      "Organization": [
        "RH"
      ],
      "OrganizationalUnit": null,
      "Locality": null,
      "Province": null,
      "StreetAddress": null,
      "PostalCode": null,
      "SerialNumber": "",
      "CommonName": "",
      "Names": null,
      "ExtraNames": null
    },
    "Subject": {
      "Country": null,
      "Organization": null,
      "OrganizationalUnit": null,
      "Locality": null,
      "Province": null,
      "StreetAddress": null,
      "PostalCode": null,
      "SerialNumber": "",
      "CommonName": "",
      "Names": null,
      "ExtraNames": null
    },
    "NotBefore": "0001-01-01T00:00:00Z",
    "NotAfter": "0001-01-01T00:00:00Z",
    "KeyUsage": 0,
    "Extensions": null,
    "ExtraExtensions": null,
    "UnhandledCriticalExtensions": null,
    "ExtKeyUsage": null,
    "UnknownExtKeyUsage": null,
    "BasicConstraintsValid": false,
    "IsCA": false,
    "MaxPathLen": 0,
    "MaxPathLenZero": false,
    "SubjectKeyId": null,
    "AuthorityKeyId": null,
    "OCSPServer": null,
    "IssuingCertificateURL": null,
    "DNSNames": null,
    "EmailAddresses": null,
    "IPAddresses": null,
    "URIs": null,
    "PermittedDNSDomainsCritical": false,
    "PermittedDNSDomains": null,
    "ExcludedDNSDomains": null,
    "PermittedIPRanges": null,
    "ExcludedIPRanges": null,
    "PermittedEmailAddresses": null,
    "ExcludedEmailAddresses": null,
    "PermittedURIDomains": null,
    "ExcludedURIDomains": null,
    "CRLDistributionPoints": null,
    "PolicyIdentifiers": null
  },
  "Reason": 1,
  "Detail": "more detail"
}
```

```suggestion
			logger.Errorf(err, "Invalid certificate error: %s", resource.ToJSON(invalidCertificateError))
```

Reply:
Right, that’s exactly what I was after, I was waiting to bring my test cluster back up today (with an expired certificate on the broker) to check. I’ve simplified the code since the error type is shown explicitly in the JSON.